D3D12Bundles/D3D12Bundles.cs

```diff
@@ -412,42 +415,41 @@ struct Vertex {
-
-                // Modify the description to use an alternate pixel shader and create
-                // a second PSO.
-                psoDesc.PixelShader = pixelShader2;
-                mPipelineState2 = mDevice.CreateGraphicsPipelineState(psoDesc);
-                mPipelineState2.Name = nameof(mPipelineState2);
-            }
+                mPipelineState1 = new PipelineState(mGraphicsDevice.NativeDevice, mRootSignature, psoDesc1);
+                mPipelineState1.NativePipelineState.Name = nameof(mPipelineState1);
 
-            // Create the command list.
-            mCommandList = mDevice.CreateCommandList<ID3D12GraphicsCommandList>(0, CommandListType.Direct, mCommandAllocator, null);
-            mCommandList.Name = nameof(mCommandList);
-
-            // Create render target views (RTVs).
-            {
-                CpuDescriptorHandle rtvHandle = mRtvHeap.GetCPUDescriptorHandleForHeapStart();
-
-                for (int i = 0; i < FrameCount; i++) {
-                    mRenderTargets[i] = mSwapChain.GetBuffer<ID3D12Resource>(i);
-
-                    mDevice.CreateRenderTargetView(mRenderTargets[i], null, rtvHandle);
-                    rtvHandle = rtvHandle.Offset(1, mRtvDescriptorSize);
-                    mRenderTargets[i].Name = $"{nameof(mRenderTargets)}[{i}]";
-                }
+                // Duplicate the description but use an alternate pixel shader and create a second PSO.
+                var psoDesc2 = new GraphicsPipelineStateDescription {
+                    //InputLayout = new InputLayoutDescription(Vertex.InputElements),
+                    InputLayout = new InputLayoutDescription(FlatShadedVertex.InputElements),
+                    RootSignature = mRootSignature,
+                    VertexShader = vertexShader,
+                    PixelShader = pixelShader2,
+                    RasterizerState = RasterizerDescription.CullNone, //I think this corresponds to CD3DX12_RASTERIZER_DESC(D3D12_DEFAULT)
+                    BlendState = BlendDescription.Opaque,  //Nothing seems to correspond to CD3DX12_BLEND_DESC(D3D12_DEFAULT)
+                    DepthStencilState = DepthStencilDescription.Default,
+                    DepthStencilFormat = mPresenter.PresentationParameters.DepthStencilFormat,
+                    SampleMask = uint.MaxValue,
+                    PrimitiveTopologyType = PrimitiveTopologyType.Triangle,
+                    RenderTargetFormats = new[] { mPresenter.PresentationParameters.BackBufferFormat, },
+                    SampleDescription = SampleDescription.Default,
+                };
+                mPipelineState2 = new PipelineState(mGraphicsDevice.NativeDevice, mRootSignature, psoDesc2);
+                mPipelineState2.NativePipelineState.Name = nameof(mPipelineState2);                
             }
 
+            // Reset the command list, we need it open for initial GPU setup.
+            mGraphicsDevice.CommandList.Reset();
```

Comment:
The GraphicsDevice which constructs this command list automatically closes it.  So we need to reset it here so we can use it for initial GPU setup, otherwise we'll hit errors.